core/overdrive.py

```diff
@@ -269,46 +289,14 @@ def _determine_hosts(self, configuration: OverdriveConfiguration) -> Dict[str, s
         # Figure out which hostnames we'll be using when constructing
         # endpoint URLs.
         server_nickname = (
-            configuration.server_nickname or OverdriveConfiguration.PRODUCTION_SERVERS
+            configuration.overdrive_server_nickname
+            or OverdriveConfiguration.PRODUCTION_SERVERS
         )
```

Comment:
We should be able to use `self._server_nickname` here, rather than duplicating the logic used calculate it.